tests/tests.go

```diff
@@ -267,13 +275,13 @@ func ChangeManagerLeadership() {
 	}, 2*time.Minute, 3*time.Second).Should(BeTrue(), "should fail with IsNotFound when kubemacpool leader pod is delete")
 
 	By("Wait for the other pod to take over leadership")
-	Eventually(func() []corev1.Pod {
-		leaderPods = corev1.PodList{}
-		testClient.VirtClient.List(context.TODO(), &leaderPods, filterByLeaderLabel)
-		return leaderPods.Items
-	}, 2*time.Minute, 3*time.Second).Should(HaveLen(1), "should have just one leader manager pod after deleting previous leader")
+	leaderPod = corev1.Pod{}
+	Eventually(func() error {
+		var err error
```

Comment:
I think you can use `:=` instead of explicitly declare error variable 
```
Eventually(func() error {
    return _, err := getLeaderPod()
}
...
```

Reply:
If I do that it will create a different instance of `leaderPod` and I want to have `leaderPod` ouside of `Eventually`